testbot.js

```diff
@@ -736,6 +722,36 @@ function handleEditCommand(msg, args) {
     });
 }
 
+function validatePriceModifier(priceModifier, finalPrice = null) {
+    /*
+    This function validates the price modifier and final price.
+    The price modifier should be a number with an optional % sign.
+    The final price should be a positive number.
+
+    Arguments:
+        priceModifier -- the price modifier to validate
+        finalPrice -- the final price to validate
+
+    Returns:
+        True if the price modifier and final price are valid, False otherwise
+    */
+
+    // remove the % sign if present
+    const priceModifierClean = priceModifier.replace('%', '');
+
+    // ensure the price modifier is a number
+    if (isNaN(priceModifierClean)) {

```

Comment:
Example of input validation - this ensures the user doesn't enter a string value causing errors further down the line,

Ideally this should be done for all user input values